atomix/cluster/src/main/java/io/atomix/cluster/messaging/MessagingService.java

```diff
@@ -70,7 +78,23 @@ CompletableFuture<Void> sendAsync(
       Address address, String type, byte[] payload, boolean keepAlive);
 
   /**
-   * Sends a message asynchronously and expects a response.
+   * Sends a message asynchronously and expects a response on a pooled connection.
+   *
+   * <p>The future may be completed exceptionally with one of:
+   *
+   * <ul>
+   *   <li>{@link IllegalStateException} - the underlying service cannot send the given request
+   *       (e.g. the service is not running)
+   *   <li>{@link java.net.ConnectException} - indicates the recipient is unreachable
+   *   <li>{@link java.util.concurrent.TimeoutException} - indicates no response came back after the
+   *       default timeout
+   *   <li>{@link io.atomix.cluster.messaging.MessagingException.ProtocolException} - indicates the
+   *       recipient failed to parse the incoming message
+   *   <li>{@link io.atomix.cluster.messaging.MessagingException.NoRemoteHandler} - indicates the
+   *       recipient received the message, but was no expecting to
+   *   <li>{@link io.atomix.cluster.messaging.MessagingException.RemoteHandlerFailure} - indicates
+   *       the recipient parsed and expected the message, but it failed unexpectedly to process it
```

Comment:
🔧 I think correct java doc would be to use throws right? 🤔 But anyway it is great you documented it 👍🏼

Reply:
I tried but then later when you build/release the javadoc, it complains that the method doesn't actually throw the exception :sweat_smile: